packages/core/src/languages/solidity/compiler.ts

```diff
@@ -104,3 +108,37 @@ export const getMinimumCompilerInput = (
 
   return minimumCompilerInput
 }
+
+/**
+ * Returns the minimum compiler output for a given source name.
+ *
+ * @param fullCompilerOutput The full compiler output object.
+ * @param fullOutputContracts The full compiler output contract object.
+ * @param sourceName The source name.
+ * @returns Minimum compiler input necessary to compile the source name.
+ */
+export const getMinimumCompilerOutput = (
```

Comment:
What's the purpose of this function?

Reply:
This may not be necessary anymore, so I'll test out removing it. 

It was needed to filter out some unnecessary artifacts that were included in the compiler output due to the user importing our foundry contract which imports our core contracts. This resulted in some dependency contracts being included which could generate errors due to the user not having the correct dependencies installed.

Reply:
Sounds good, lmk if you want to keep it in and i'll review this function

Reply:
So I double-checked this and it is required whenever the foundry library contract is included in the compilation process because it causes all of our system contracts to be compiled. Then some of our system dependencies are filtered out by the `getMinimumCompilerInput` function which causes an error when we try to create the `UpgradableContract` with them. So this solution handles that by filtering everything from both the compiler input and output. 

An alternative solution would be to not use the `getMinimumCompilerInput` function at all, or update it to avoid filtering out any of our system contracts or their dependencies. But IMO, the current solution of using a mirror `getMinimumCompilerOutput` function seems better to me. So yeah, you should review it.

Reply:
LGTM